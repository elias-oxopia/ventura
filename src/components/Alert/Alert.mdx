---
name: Alert
menu: Components
route: /alert
---

import { Playground, Props } from 'docz';
import { Alert } from '../..';

# Alert

Alerts allow you to display error messages relevant to an entire form or application.

## Examples

### Basic usage

<Playground>
  <Alert intent='error' message='Something went wrong.' />
</Playground>

## API

<Props of={Alert} />
