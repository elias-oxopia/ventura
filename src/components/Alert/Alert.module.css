.alert {
  padding: 14.5px 18px;
  display: flex;
  align-items: center;
  border-radius: var(--border-radius-small);
}

.alert.alertError {
  background-color: var(--color-error-light);
}

.alert.alertSuccess {
  background-color: var(--color-success-light);
}

.message {
  display: inline-block;
}

svg.icon {
  margin-right: 10px;
  height: 16px;
  width: 16px;
}

svg.icon.iconSuccess {
  color: var(--color-success);
}

svg.icon.iconError {
  color: var(--color-error);
}
