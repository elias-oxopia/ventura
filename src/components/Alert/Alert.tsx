import React from 'react';
import { AlertCircle } from 'react-feather';
import classNames from 'classnames';

import cssReset from '../../css-reset.module.css';
import styles from './Alert.module.css';

interface Props {
  intent?: 'error';
  message: string;
  className?: string;
}

const Alert: React.FC<Props> = ({ intent = 'error', message, className }: Props) => {
  const mergedClassNames = classNames(cssReset.ventura, styles.alert, className);

  return (
    <div className={mergedClassNames}>
      {intent === 'error' && <AlertCircle className={styles.icon} />}
      <span className={styles.message}>{message}</span>
    </div>
  );
};

export default Alert;
