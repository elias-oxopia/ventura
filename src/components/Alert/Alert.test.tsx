import React from 'react';
import { render } from '@testing-library/react';

import Alert from './Alert';

describe('Input', () => {
  const defaultProps = {
    message: 'Something went wrong.',
  };

  test('default snapshot', () => {
    const { asFragment } = render(<Alert {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('the message should be rendered', () => {
    const { queryByText } = render(<Alert {...defaultProps} />);
    const alertElement = queryByText(/Something went wrong./);
    expect(alertElement).not.toBe(null);
  });

  test('className prop', () => {
    const className = 'center';
    const { getByText } = render(<Alert {...defaultProps} className={className} />);
    const alertElement = getByText(/Something went wrong./).parentElement?.parentElement!;

    const renderedClassNames = alertElement.className.split(' ');
    expect(renderedClassNames).toContain(className);
    // className in prop should be the last in the row
    expect(renderedClassNames.indexOf(className)).toBe(renderedClassNames.length - 1);
  });
});
