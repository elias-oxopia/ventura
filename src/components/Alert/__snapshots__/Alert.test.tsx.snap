// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input default snapshot 1`] = `
<DocumentFragment>
  <div
    class="ventura alert alertError"
  >
    <div
      class="contentContainer"
    >
      <svg
        class="icon iconError"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="7.86 2 16.14 2 22 7.86 22 16.14 16.14 22 7.86 22 2 16.14 2 7.86 7.86 2"
        />
        <line
          x1="15"
          x2="9"
          y1="9"
          y2="15"
        />
        <line
          x1="9"
          x2="15"
          y1="9"
          y2="15"
        />
      </svg>
      <span
        class="message"
      >
        Something went wrong.
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input error snapshot 1`] = `
<DocumentFragment>
  <div
    class="ventura alert alertError"
  >
    <div
      class="contentContainer"
    >
      <svg
        class="icon iconError"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="7.86 2 16.14 2 22 7.86 22 16.14 16.14 22 7.86 22 2 16.14 2 7.86 7.86 2"
        />
        <line
          x1="15"
          x2="9"
          y1="9"
          y2="15"
        />
        <line
          x1="9"
          x2="15"
          y1="9"
          y2="15"
        />
      </svg>
      <span
        class="message"
      >
        Something went wrong.
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input success snapshot 1`] = `
<DocumentFragment>
  <div
    class="ventura alert alertSuccess"
  >
    <div
      class="contentContainer"
    >
      <svg
        class="icon iconSuccess"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
        />
        <polyline
          points="22 4 12 14.01 9 11.01"
        />
      </svg>
      <span
        class="message"
      >
        Something went wrong.
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Input warning snapshot 1`] = `
<DocumentFragment>
  <div
    class="ventura alert alertWarning"
  >
    <div
      class="contentContainer"
    >
      <svg
        class="icon iconWarning"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
        />
        <line
          x1="12"
          x2="12"
          y1="9"
          y2="13"
        />
        <line
          x1="12"
          x2="12.01"
          y1="17"
          y2="17"
        />
      </svg>
      <span
        class="message"
      >
        Something went wrong.
      </span>
    </div>
  </div>
</DocumentFragment>
`;
